org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultModelProvider.java

```diff
@@ -1,79 +0,0 @@
-/*
- * Copyright (c) 2016 the original author or authors.
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- */
-package org.eclipse.buildship.core.workspace.internal;
-
-import java.util.List;
-import java.util.Set;
-
-import org.gradle.tooling.CancellationToken;
-import org.gradle.tooling.GradleConnector;
-import org.gradle.tooling.ProgressListener;
-
-import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableList;
-
-import com.gradleware.tooling.toolingmodel.OmniBuildEnvironment;
-import com.gradleware.tooling.toolingmodel.OmniEclipseProject;
-import com.gradleware.tooling.toolingmodel.OmniGradleBuild;
-import com.gradleware.tooling.toolingmodel.OmniGradleProject;
-import com.gradleware.tooling.toolingmodel.repository.FetchStrategy;
-import com.gradleware.tooling.toolingmodel.repository.ModelRepository;
-import com.gradleware.tooling.toolingmodel.repository.TransientRequestAttributes;
-
-import org.eclipse.core.runtime.IProgressMonitor;
-
-import org.eclipse.buildship.core.CorePlugin;
-import org.eclipse.buildship.core.console.ProcessStreams;
-import org.eclipse.buildship.core.util.progress.DelegatingProgressListener;
-import org.eclipse.buildship.core.workspace.ModelProvider;
-
-/**
- * Default implementation of {@link ModelProvider}.
- *
- * @author Stefan Oehme
- */
-final class DefaultModelProvider implements ModelProvider {
```

Comment:
This should still exist and take care of caching. For now I'd use exactly the same strategy (FetchStrategy enum passed as an argument) as before. We can improve that later, it would blow up the size of this change too much.